packages/docs/concepts/lowdefy-schema.yaml

```diff
@@ -273,15 +264,17 @@ _ref:
 
             ## References and templates
 
-            References are used to split the configuration of an app into logically distinct files, and to reuse configuration in the app. References can be used almost anywhere in the configuration, as long as the configuration remains valid YAML.
+            References are used to split the configuration of an app into logically distinct files, and to reuse configuration in the app. Configuration is primarily split up for two reasons, to improve readability or for reusability, for example, when used as templates. References can be used anywhere in the configuration, as long as the configuration resolves to comply with the Lowdefy app schema.
```

Comment:
When using references there are the following cases:
- Just splitting out the config for readabilty and code navigation
- Splitting out a piece of config because you want to use the exact same config in multiple places (code reuse level 1)
- Using a ref with variables so you can have a shared piece of config with some dynamic variables (code reuse level 2)
- Using an njk file which allows for much more complicated things - this is an advanced feature and easily overused (code reuse level 3).
- Using transformers - even more compplicated than njk files - should be used if justified like building config from a JSONSchema file (code reuse level 4).

I think the current explanation does not elaborate on the level 1 and 2 enough, and it feels like we jump directly to nunjucks templates. I also don't know if `templates` is the right word for level 1 and 2 code reuse, becuase of the confusion with nunjucks.

Reply:
Should we go into more detail on that here or on the References and Templates concept page?

Reply:
I like the breakdown @SamTolmay. I'd consider listing this description on both pages? It's important to know it exists, and can be hard to come to this realization when you are building apps. We're also not making it easy since the same tool / named thing can be used to solve all these probelms.

Reply:
Made some changes to this section in 5837c29b0e95dd7432b8f4c6d314179c9aaf1428

Reply:
> Made some changes to this section in [5837c29](https://github.com/lowdefy/lowdefy/commit/5837c29b0e95dd7432b8f4c6d314179c9aaf1428)

Made the same changes in 1ee298ee370bac30e1189237d5cf9ccb69a5fac3